### PR TITLE
Fix #1646:Changed all padding_# and margin_# to space_#dp in dimens value.

### LINKED ISSUE #1646: Make dimens value follow one naming convention
All `dimens` in app modules should follow below mentioned convention:

1_66JEKhu7WqtPMkjegXMmgQ

In this image, I suggest instead of using `padding_24dp` and `margin_24dp`, combining them to `space_24dp`.
So basically all padding/margins should be combined to `space_<value>`. This was we will have to add lesser values to dimens file.

[Image Reference](https://medium.com/mindorks/android-resource-naming-convention-42e4e8026614)

### PR DESCRIPTION
**Pull request by ranjsa**
## Explanation

Fixes [#1646 ](https://github.com/oppia/oppia-android/issues/1646)

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/layout-land/walkthrough_final_fragment.xml [hunk 1]
```diff
@@ -43,12 +43,12 @@
...
         android:layout_width="0dp"
         android:layout_height="wrap_content"
-        android:layout_marginEnd="@dimen/margin_8"
+        android:layout_marginEnd="@dimen/space_8dp"
         android:layout_weight="1"
         android:background="?attr/selectableItemBackground"
         android:onClick="@{(v) -> presenter.goBack()}"
         app:cardCornerRadius="4dp"
-        app:contentPadding="@dimen/padding_24dp">
+        app:contentPadding="@dimen/space_24dpdp">
```
Comment (anandwana001) [inline app/src/main/res/layout-land/walkthrough_final_fragment.xml]: it should be `space_24dp` not `dpdp`. 

here and all places.

File: app/src/main/res/layout-land/walkthrough_final_fragment.xml [hunk 2]
```diff
@@ -75,13 +75,13 @@
...
         android:layout_height="match_parent"
-        android:layout_marginStart="@dimen/margin_8"
+        android:layout_marginStart="@dimen/space_8dp"
         android:layout_weight="1"
         android:background="?attr/selectableItemBackground"
         android:clickable="true"
         android:focusable="true"
         app:cardCornerRadius="4dp"
-        app:contentPadding="@dimen/padding_24dp">
+        app:contentPadding="@dimen/space_24dpdp">
```
Comment (anandwana001) [inline app/src/main/res/layout-land/walkthrough_final_fragment.xml]: ditto

File: app/src/main/res/layout/walkthrough_final_fragment.xml [hunk 3]
```diff
@@ -48,12 +48,12 @@
...
           android:layout_width="0dp"
           android:layout_height="wrap_content"
-          android:layout_marginEnd="@dimen/margin_8"
+          android:layout_marginEnd="@dimen/space_8dp"
           android:layout_weight="1"
           android:background="?attr/selectableItemBackground"
           android:onClick="@{(v) -> presenter.goBack()}"
           app:cardCornerRadius="4dp"
-          app:contentPadding="@dimen/padding_24dp">
+          app:contentPadding="@dimen/space_24dpdp">
```
Comment (anandwana001) [inline app/src/main/res/layout/walkthrough_final_fragment.xml]: ditto

File: app/src/main/res/layout/walkthrough_final_fragment.xml [hunk 4]
```diff
@@ -80,13 +80,13 @@
...
           android:layout_height="wrap_content"
-          android:layout_marginStart="@dimen/margin_8"
+          android:layout_marginStart="@dimen/space_8dp"
           android:layout_weight="1"
           android:background="?attr/selectableItemBackground"
           android:clickable="true"
           android:focusable="true"
           app:cardCornerRadius="4dp"
-          app:contentPadding="@dimen/padding_24dp">
+          app:contentPadding="@dimen/space_24dpdp">
```
Comment (anandwana001) [inline app/src/main/res/layout/walkthrough_final_fragment.xml]: ditto

File: app/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -14,58 +13,42 @@
   <dimen name="elevation_4dp">4dp</dimen>
   <dimen name="elevation_0dp">0dp</dimen>
   <dimen name="padding_296dp">296dp</dimen>
-  <dimen name="padding_72dp">72dp</dimen>
```
Comment (anandwana001) [inline app/src/main/res/values/dimens.xml:17]: @rt4914 Do we have any specific difference in `padding_72dp ` vs `padding_72 `?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Nice work @ranjsa 
Let's wait for Rajat comment on dpdp thing and then you can update it accordingly. 

Added a few points and also added 1 point to clarify between `padding_72` vs `padding_72dp`.

**Comment by ranjsa** [comment]
> Nice work @ranjsa
> Let's wait for Rajat comment on dpdp thing and then you can update it accordingly.
> 
> Added a few points and also added 1 point to clarify between `padding_72` vs `padding_72dp`.

Thank You @anandwana001 for reviewing my PR. I have fixed all space_#dpdp to space_#dp.

File: app/src/main/res/layout/walkthrough_final_fragment.xml (hunk 3 above)
Comment (ranjsa) [reply app/src/main/res/layout/walkthrough_final_fragment.xml]: done

File: app/src/main/res/layout-land/walkthrough_final_fragment.xml (hunk 2 above)
Comment (ranjsa) [reply app/src/main/res/layout-land/walkthrough_final_fragment.xml]: done

File: app/src/main/res/layout-land/walkthrough_final_fragment.xml (hunk 1 above)
Comment (ranjsa) [reply app/src/main/res/layout-land/walkthrough_final_fragment.xml]: done

File: app/src/main/res/values/dimens.xml [hunk 6]
```diff
@@ -14,58 +13,40 @@
   <dimen name="elevation_4dp">4dp</dimen>
   <dimen name="elevation_0dp">0dp</dimen>
   <dimen name="padding_296dp">296dp</dimen>
```
Comment (anandwana001) [inline app/src/main/res/values/dimens.xml]: let's update this padding too.

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/dimens.xml (hunk 6 above)
Comment (anandwana001) [reply app/src/main/res/values/dimens.xml]: @ranjsa I think we are not using this padding anywhere, could you please verify this and remove if not needed?

Comment (ranjsa) [reply app/src/main/res/values/dimens.xml]: yes @anandwana001 we were not using this anywhere so I removed it

**Review by anandwana001 — APPROVED** [review]
LGTM, Thanks @ranjsa 
@rt4914 can take a quick pass on this and then we are good to merge this. By the time, feel free to check other issues to work on.

**Comment by ranjsa** [comment]
> LGTM, Thanks @ranjsa
> @rt4914 can take a quick pass on this and then we are good to merge this. By the time, feel free to check other issues to work on.

Okay sure, Thanks @anandwana001

**Comment by BenHenning** [comment]
Ideally, we should be semantically naming all dimensions. @rt4914 would you prefer that we check this PR in as-is and fix that in a later change?

**Review by rt4914 — APPROVED** [review]
Thanks  a lot @ranjsa . This is really great.

Also, I understand that its not easy to make a change and resolve merge conflicts again and again and therefore I am approving this so that if you want to merge this PR at this stage you can do that.

If you choose to make the suggested change, then just assign it back to me and I review back and merge.

If you choose to merge it right now, then I suggest you file an issue [here](https://github.com/oppia/oppia-android/issues/new) . You can also add this under SLoP tags too.

Sounds good?

**Comment by rt4914** [comment]
> Ideally, we should be semantically naming all dimensions. @rt4914 would you prefer that we check this PR in as-is and fix that in a later change?

@BenHenning I am not 100% sure what you mean here?
(a.) Is it about reordering or the `space_` values in `dimens.xml` file
or (b.) The naming should be changed to provide more context to paddings/margins?
or something else.

**Comment by ranjsa** [comment]
Hello, @rt4914 thanks you for reviewing this PR.
Also, did you mean  creating a new issue for this https://github.com/oppia/oppia-android/pull/1936#issuecomment-704559424 
https://github.com/oppia/oppia-android/pull/1936#issuecomment-704996261 if so I would like to work on this issue on a different PR if I am right then please confirm once so I can proceed.

**Comment by rt4914** [comment]
> Hello, @rt4914 thanks you for reviewing this PR.
> Also, did you mean creating a new issue for this [#1936 (comment)](https://github.com/oppia/oppia-android/pull/1936#issuecomment-704559424)
> [#1936 (comment)](https://github.com/oppia/oppia-android/pull/1936#issuecomment-704996261) if so I would like to work on this issue on a different PR if I am right then please confirm once so I can proceed.

Yeah, you should file a new issue.
I will merge this PR now and you can file a new issue and start working on it.

**Pull request merged**